icevision/models/ross/efficientdet/dataloaders.py

```diff
@@ -92,26 +92,22 @@ def build_train_batch(records, batch_tfms=None):
     ```
     """
     records = common_build_batch(records, batch_tfms=batch_tfms)
+    batch_images, batch_bboxes, batch_classes = zip(
+        *(process_train_record(record) for record in records)
+    )
 
-    images = []
-    targets = {"bbox": [], "cls": []}
-    for record in records:
-        image = im2tensor(record["img"])
-        images.append(image)
-
-        if len(record["labels"]) == 0:
-            targets["cls"].append(tensor([0], dtype=torch.float))
-            targets["bbox"].append(tensor([[0, 0, 0, 0]], dtype=torch.float))
-        else:
-            labels = tensor(record["labels"], dtype=torch.float)
-            targets["cls"].append(labels)
-
-            bboxes = tensor([bbox.yxyx for bbox in record["bboxes"]], dtype=torch.float)
-            targets["bbox"].append(bboxes)
+    # convert to tensors
+    batch_images = torch.stack(batch_images)
+    batch_bboxes = [tensor(bboxes, dtype=torch.float32) for bboxes in batch_bboxes]
```

Comment:
It's a bit more verbose, but I like the fact that we're passing `dtype` explicitly here, specially because as you pointed out `effdet` complains if the dtype is not what it expects